superset/assets/javascripts/explorev2/stores/fields.js

```diff
@@ -993,15 +1004,15 @@ export const fields = {
   },
 
   viewport_latitude: {
-    type: 'IntegerField',
+    type: 'TextField',
```

Comment:
I figured that was the intent and had some thoughts about how we may want to express that in the future. For now I wanted to avoid the "virtual fields" that had no direct corresponding components.

I'm thinking that controls configuration should allow for a `validators` prop in the form of an array of functions. We can define a set of common validators (validators.integer, validators.non_null, validators.at_least_one), process them in the FieldSet controller, and give cues to the users (maybe a red exclamation point).

The validator function would receive the `value`, the field configuration, and be expected to return a comprehensive message. Complex fields will be allowed to define complex validators if needed.